### DISCUSSION TITLE
Authenticating using SAML Okta

### DISCUSSION THREAD

**Discussion by midiaxe** [pspete/psPAS#531] [General]
Hello, I'm trying to authenticate using SAML Okta, see below.  For some reason, I'm getting an error message on line 2.  I'm able to get a valid login response from SAML but PASSession failed to authenticate.  Using PVWA UI, I'm able to log in using Okta with no issues.  Just wondering if anybody has ideas on how to troubleshoot this.  I appreciate any help.

Code:
`$loginResponse = New-SAMLInteractive -LoginIDP $loginURL
New-PASSession -SAMLAuth -concurrentSession $true -BaseURI $baseURL -SAMLResponse $loginResponse
`

Error message:
```
Invoke-PASRestMethod : [400] Authentication failure. Please contact your Administrator.
At line:652 char:21
+ ...                      $PASSession = Invoke-PASRestMethod @LogonRequest
+                                        ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : NotSpecified: ({"ErrorCode":"P...dministrator."}:ErrorRecord) [Invoke-PASRestMethod], Exception
    + FullyQualifiedErrorId : PASWS035E,Invoke-PASRestMethod

```

**Comment by pspete** [comment]
you can try and authenticate without using the module and see if it succeeds or not:
```powershell
$loginResponse = New-SAMLInteractive -LoginIDP $loginURL
$LogonRequest = @{ }
$Uri = "https://pvwa-company/PasswordVault"

$LogonRequest['Method'] = 'POST'
$LogonRequest['Body'] = @{'SAMLResponse' = $loginResponse ; 'apiUse' = $true } | ConvertTo-Json
$LogonRequest['ContentType'] = 'application/x-www-form-urlencoded'
$LogonRequest['Uri'] = "$Uri/api/auth/SAML/Logon"

Invoke-RestMethod @LogonRequest
```

**Reply by midiaxe** [reply]
Using the code above, it was successful.

**Reply by pspete** [reply]
Thats the exact code the module uses....
We can't replicate the error you see when using the module
<img width="710" alt="image" src="https://github.com/pspete/psPAS/assets/10600090/7b5a44eb-12fd-4d02-b5f8-92bddfca3188">
